### DISCUSSION TITLE
Need Help Integrating Zustand for State Management Inside and Outside React Components

### DISCUSSION THREAD

**Discussion by talpx0** [pmndrs/zustand#2363] [Q&A]
---

Hello, fellow developers!

I'm working on a project where I need to manage global state accessible both inside React components and in external modules (like the React Expo Task Manager). My goal is to create a Zustand store that allows for state mutations in both contexts seamlessly.

I started with the basic setup of Zustand and managed to create a store using `zustand/vanilla` for outside React components, but I've hit a roadblock when trying to mutate this state from within React components. Here's a simplified version of what I've done:

```javascript
import create from 'zustand'

// Define the store
const useDogStore = create(() => ({
  paw: true,
  snout: true,
  fur: true,
}))

// Outside component mutation works fine
useDogStore.setState({ paw: false })

// Inside a React component, trying to read and mutate state
function DogComponent() {
  const paw = useDogStore(state => state.paw)
  // Attempting any mutation here leads to an infinite loop
}
```

While I can mutate the state from outside React components without issues, I'm still able to read it in component , but  attempting to mutate it from within react component causes infinite loops. I suspect the issue arises when trying to update the state inside a component, but I can't pinpoint what I'm doing wrong.

Has anyone faced a similar issue or can offer guidance on how to properly integrate Zustand for this use case? Any advice on avoiding infinite loops while allowing state mutations both inside and outside React components would be greatly appreciated!

Thank you in advance for your help!

---

**Comment by talpx0** [comment]
fixed , it just very simple , the document is not clear  .

**Comment by talpx0** [comment]
fixed , it just very simple , the document is not clear  .

**Reply by dbritto-dev** [reply]
@talpx0 I don't see any error on your example, what's your issue?
